src/main/java/servlets/DataEntries.java

```diff
@@ -61,24 +57,26 @@ private String getJson() {
       return "{}";
     }
 
+    // Associate a runID with a list (for json) of the entry types that are found
+    // for it.
     HashMap<String, TreeSet<String>> dataMap = new HashMap<String, TreeSet<String>>();
     for (File file : dataEntries) {
-      // Parse through all entries and write them as JSON
+      // Parse through all entries and write them as JSON.
       String name = file.getName();
-      System.out.println("The filename is " + name);
+      // Use regex to extract name and type from string.
       String[] params = name.split("[_.]");
       if (params.length != 3) {
         System.err.println("Invalid database entry format!");
         return "";
       }
-      String hash = params[0];
+      String runId = params[0];
       String type = params[1];
-      if (!dataMap.containsKey(hash)) {
+      if (!dataMap.containsKey(runId)) {
         TreeSet<String> toAdd = new TreeSet<String>();
-        toAdd.add(type);
-        dataMap.put(hash, toAdd);
+        toAdd.add(runId);
+        dataMap.put(runId, toAdd);
       } else {
-        dataMap.get(hash).add(type);
+        dataMap.get(runId).add(type);
```

Comment:
I think you could simplify this with `HashMap.computeIfAbsent`.

```
dataMap.computeIfAbsent(runId, id -> new TreeSet<String>()).add(type);
```

Reply:
> Can you put the new homepage stuff in a separate PR? It's not closely related to the servlets, and reviews will go quicker if the PR doesn't keep growing.

Some of the initial comments on the pr were regarding features in the home page, naming conventions etc, which is why I committed these features into this branch. Should I not address those comments then? I definitely think it makes sense to have it a separate pr, but also it doesn't make sense to change things I've already written while working on this branch?

Reply:
Ah, I didn't remember that. It's good to address comments, but try not to add new features to PRs in review. I think "Add searchable feature" is what added most of the new code since we started review.